app/actions/handlers.py

```diff
@@ -21,8 +21,8 @@ async def action_auth(integration, action_config: AuthenticateConfig):
 
 async def action_fetch_samples(integration, action_config: PullObservationsConfig):
     logger.info(f"Executing fetch_samples action with integration {integration} and action_config {action_config}...")
-    vehicles = {
-        "observations_extracted": [
+    observations = {
+        "observations_extracted": 
```

Comment:
I think that `observations` could just be the list of objects, without needing a dict with "observations_extracted".